azure/datalake/store/core.py

```diff
@@ -865,6 +865,8 @@ def read(self, length=-1):
             data_read = self.cache[self.loc - self.start:
                              min(self.loc - self.start + length, self.end - self.start)]
             out += data_read
+            if len(data_read) == 0: # Check to catch possible server errors. Ideally shouldn't happen.
```

Comment:
For what I understand, `data_read` is an array. Then the recommended way to test if it's empty is `if not data_read`:
https://stackoverflow.com/questions/43121340/why-is-the-use-of-lensequence-in-condition-values-considered-incorrect-by-pyli

Reply:
Changed it to not(data_read)